lib/hanami/middleware/body_parser/class_interface.rb

```diff
@@ -9,15 +9,38 @@ class BodyParser
       # @api private
       # @since 1.3.0
       module ClassInterface
+        # Instantiate a new body parser instance and load its parsers
+        #
+        # @example
+        #   Hanami::Middleware::BodyParser.new(->(env) { [200, {}, "app"] }, :json)
+        #
+        #   Hanami::Middleware::BodyParser.new(
+        #     ->(env) { [200, {}, "app"] }, [json: "application/json+scim"]
+        #   )
+        #
+        #   Hanami::Middleware::BodyParser.new(
+        #     ->(env) { [200, {}, "app"] }, [json: ["application/json+scim", "application/ld+json"]]
+        #   )
```

Comment:
This example helps so much, thank you @solnic!!

Just so I can understand, is this what the `parser_specs` arg would look like if we wanted to configure multiple parsers with custom mime types?

```ruby
[
  {json: ["application/json+scim", "application/ld+json"]},
  {xml: ["application/xml+foo", "application/xml+bar"]}
]
```

And am I right in assuming that this is what we'd do if we wanted multiple parsers without any custom mime types?

```ruby
[:json, :xml]
```

Reply:
(I actually wonder if the case of multiple parsers with multiple custom mime types would be good to include in the tests)

Reply:
Actually, does all our Array() wrapping allow the parser_specs to be passed like this?

```ruby
Hanami::Middleware::BodyParser.new(env, json: ["custom", "type"], xml: ["custom", "types"])
```

Because that would be ideal. It feels the most direct and natural way to express it.

Reply:
@timriley the intention was to support what's passed from the config, I assumed nobody will be initializing a body parser manually, it's an implementation detail and the whole class is actually private. It doesn't make sense to focus on its ergonomics because of this. I would change many things here but it's just not worth the effort.

> Actually, does all our Array() wrapping allow the parser_specs to be passed like this?
> 
> Hanami::Middleware::BodyParser.new(env, json: ["custom", "type"], xml: ["custom", "types"])

I've made this possible:

```ruby
Hanami::Middleware::BodyParser.new(env, [json: ["custom", "type"], xml: ["custom", "types"]])
```

Which would be passed in if you did this in the config:

```ruby
config.middleware.use(:body_parser, [json: ["custom", "type"], xml: ["custom", "types"]])
```

FYI the reason why this is needed to be passed in as an array is because `use` is a method defined in `Middleware::Stack` which supports its own options. This kind of sucks and it was mentioned when I introduced these options (`before` and `after`) but I'm not sure if it makes sense to do anything about it now.